src/lib/src/x86_calcbonds.cpp

```diff
@@ -29,45 +27,40 @@ constexpr std::size_t kBigVectorThreshold = 256 * 1024;
 // faster for bigger tasks (saving one read for every write).
 constexpr std::size_t kStreamingThreshold = 16 * 1024 * 1024;
 
-template <bool streaming_store, typename VectorT>
-void CalcBondsOrthoX86Vec(const VectorToScalarT<VectorT> *coords0,
-                          const VectorToScalarT<VectorT> *coords1,
-                          const VectorToScalarT<VectorT> *box, std::size_t n,
-                          VectorToScalarT<VectorT> *out) {
-  VectorToScalarT<VectorT> boxx = box[0], boxy = box[1], boxz = box[2];
-  // generate packed AOS box structs
-  auto box_packed = VectorTriple<VectorT>();
-  Interleave3(boxx, boxy, boxz, box_packed.x, box_packed.y, box_packed.z);
-  // box_packed.x = xyzx box_packed.y = yzxy box_packed.z = zxyz
+template <bool streaming_store, typename VectorT, typename BoxT>
+void CalcBondsInner(const VectorToScalarT<VectorT> *coords0,
+                    const VectorToScalarT<VectorT> *coords1,
+                    const VectorToScalarT<VectorT> *box,
+                    std::size_t n,
+                    VectorToScalarT<VectorT> *out) {
+  auto vecbox = BoxT(box);
 
   size_t i = 0;
-  for (; distopia_unlikely(!IsAligned<VectorT>(&out[i]) && i < n); ++i) {
-    auto x0 = coords0[3 * i], x1 = coords1[3 * i];
-    auto y0 = coords0[3 * i + 1], y1 = coords1[3 * i + 1];
-    auto z0 = coords0[3 * i + 2], z1 = coords1[3 * i + 2];
-    out[i] = Distance3DWithBoundary(x0, y0, z0, x1, y1, z1, boxx, boxy, boxz);
-  }
-  for (; i + ValuesPerPack<VectorT> - 1 < n; i += ValuesPerPack<VectorT>) {
-    // load coords in AOS
-    auto c0 = VectorTriple<VectorT>(&coords0[3 * i]);
-    auto c1 = VectorTriple<VectorT>(&coords1[3 * i]);
-    // do 1D distances in AOS, saves deinterleaving 2 position vectors in this
-    auto d_abc = Distance1DWithBoundary(c0, c1, box_packed);
-    auto d_xyz = d_abc.deinterleave();
-    // deinterleave and do hypot accumulation in SOA
-    auto res = Hypot(d_xyz.x, d_xyz.y, d_xyz.z);
+  if (n % ValuesPerPack<VectorT>) {
+    auto c0 = VectorTriple<VectorT>(coords0);
+    auto c1 = VectorTriple<VectorT>(coords1);
+
+    VectorT result = NewDistance3dWithBoundary(c0, c1, vecbox);
+
     if constexpr (streaming_store) {
```

Comment:
`if constexpr`.. again its useful here.

Reply:
Tbh I still want to see if there's any timing different between these two routes and trim one out